.github/workflows/wait-for-circleci-workflow-status.yml

```diff
@@ -13,8 +13,13 @@ jobs:
           OWNER: ${{ github.repository_owner }}
           REPOSITORY: ${{ github.event.repository.name }}
           BRANCH: ${{ github.head_ref || github.ref_name }}
+          # For a `push` event, the HEAD commit hash is `github.sha`.
+          # For a `pull_request` event, `github.sha` is instead the base branch commit hash. The
+          # HEAD commit hash is `pull_request.head.sha`.
+          HEAD_COMMIT_HASH: ${{ github.event.pull_request.head.sha || github.sha }}
         run: |
-          pipeline_id=$(curl --silent "https://circleci.com/api/v2/project/gh/$OWNER/$REPOSITORY/pipeline?branch=$BRANCH" | jq -r ".items[0].id")
+          pipeline_id=$(curl --silent "https://circleci.com/api/v2/project/gh/$OWNER/$REPOSITORY/pipeline?branch=$BRANCH" | jq -r ".items | map(select(.vcs.revision == \"${HEAD_COMMIT_HASH}\" )) | first | .id")
```

Comment:
Just a note - what happens if a CircleCI pipeline was never launched for the specified commit? The current `jq` query would return null in that case. Not sure if this edge case is handled elsewhere in the workflow, but wanted to raise it.

Reply:
It's certainly not handled explicitly, but yes it is handled in a sense, both before and after.

If `pipeline_id` is `null`, then the attempt to get `workflow_status` will be unsuccessful, resulting in `workflow_status` being set to `null` as well. The check to compare `workflow_status` to the string "running" will trigger, and the workflow will fail saying that the `workflow_status` is null.

Not ideal, but not terrible as a temporary measure (this workflow is just to help us transition from CircleCI to GitHub Actions).